packages/services-soap/src/mode/seed.ts

```diff
@@ -48,14 +54,42 @@ export class SoapClientSupplierSeed extends SoapClientSupplierBase {
     endpoint: string
   ) {
     const seedify = async (fnc: ClientFunction, name: string, args: any) => {
-      const seed = FxSt.createByUrl(`${endpoint}/${name}`, args, this.testScope)
-      const resp = await fnc(args)
-      seed.set(resp)
-      return resp
+      const seed = FxSt.createByUrl<ISoapResponse>(
+        `${endpoint}/${name}`,
+        args,
+        this.testScope
+      )
+      try {
+        const resp = await fnc(args)
+        seed.set(resp)
+        return resp
+      } catch (e) {
+        this.handleError(seed, e)
+      }
     }
     return await createProxy(client, seedify, true)
   }
 
+  private handleError(seed: FxSt<any>, error: any) {
+    const isSoapFault =
+      error &&
+      error.cause &&
+      error.cause.root &&
+      error.cause.root.Envelope &&
+      error.cause.root.Envelope.Body &&
+      error.cause.root.Envelope.Body.Fault
+    if (isSoapFault) {
+      const code = error.cause.root.Envelope.Body.Fault.faultcode
+      const description = error.cause.root.Envelope.Body.Fault.faultstring
+      const details = error.cause.root.Envelope.Body.Fault.detail
```

Comment:
You alreardy have the Fault object in isSoapFault. Use it ^^

Reply:
true